### PR TITLE
Fixed "connection already closed" error in the Registry handler.

### PR DESCRIPTION
**Pull request by ipanova**
closes #8672

### CODE REVIEW

File: pulp_container/app/registry.py [hunk 1]
```diff
@@ -97,6 +97,8 @@ async def get_tag(self, request):
                 streamed back to the client.
 
         """
+        self._reset_db_connection()
```
Comment (ipanova) [inline pulp_container/app/registry.py:100]: I was not sure where to reset the connection - directly in the plugin or core.
What would be the implications if we'd add `self._reset_db_connection()` instead in core directly in each method?
@daviddavis @mdellweg ?

**Comment by mdellweg** [comment]
I'm not sure what is happening here. As far as i have understood, django orm is recreating the connection whenever it seems to need a new one.
OTOH, i have seen a similar error message when forking the process and then closing the connection in one process while using it in the other. My suspicion is, that we may be hitting some async related bad sharing of a psycopg connection.
Do we know, where the connection is closed in the first place?

File: pulp_container/app/registry.py (hunk 1 above)
Comment (daviddavis) [reply pulp_container/app/registry.py:100]: I think what you have here is correct (at least according to my original intent, which was that it should be called once at each entry point and not in helper functions such as `_match_distribution`). 

However, I was very leery about introducing this function originally and still not sure if it's a good idea. Having someone else who's familiar with the content app to review and weigh in would be good.

**Comment by daviddavis** [comment]
Here is the original issue: https://pulp.plan.io/issues/6045

And commit: https://github.com/pulp/pulpcore/commit/5aabb202dd59cbe2d30ef5ad91f01932c9ca041b

Maybe there's some more insight there.

**Comment by mdellweg** [comment]
I also found this: https://www.psycopg.org/docs/advanced.html#support-for-coroutine-libraries

File: pulp_container/app/registry.py (hunk 1 above)
Comment (mdellweg) [reply pulp_container/app/registry.py:100]: I'd prefer to have that call being the first thing when a new request is handled. But i don't even see where in `Handler` that would be. But maybe it's safe and not a bit performance impact if we add it in more places.

**Comment by ipanova** [comment]
@dralley @bmbouter  can you provide any input on this PR please? I am going to backport and include it  into an older release needed for katello tomorrow

File: pulp_container/app/registry.py (hunk 1 above)
Comment (bmbouter) [reply pulp_container/app/registry.py:100]: I see the core's content app has these two primary entry points to the code https://github.com/pulp/pulpcore/blob/master/pulpcore/content/__init__.py#L54-L55 and I see both of them already call the `self._reset_db_connection` so I think core's already got it. Agreed that I'm also a little unclear on why it's needed in the first place, but core has it.

I'm not very familiar with pulp_container's registry, but it seems like it adds these top-level new routes: https://github.com/pulp/pulp_container/blob/master/pulp_container/app/content.py#L8-L14

Comment (ipanova) [reply pulp_container/app/registry.py:100]: pulp_container does not call into core's `stream_content` hence I needed to add `self._reset_db_connection` before `_math_distribution` is called.

**Comment by ipanova** [comment]
@mdellweg if you don't have any objections please give me an ack so i can merge and backport this

File: pulp_container/app/registry.py (hunk 1 above)
Comment (dralley) [reply pulp_container/app/registry.py:100]: I think in general though, if pulpcore is already calling it in these handlers, then adding the calls here is unlikely to break anything.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**